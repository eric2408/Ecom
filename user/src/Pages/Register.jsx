import React, { useState } from 'react'
import styled from "styled-components";
import { mobileScreen } from '../Helper';
import {Link} from 'react-router-dom';
import { generalRequest } from "../requestAxios";
import { useNavigate } from 'react-router-dom';

const Container = styled.div`
  width: 100vw;
  height: 100vh;
  background-size: cover;
  display: flex;
  justify-content: center;
  background-color: #14141f;
`;

const Padding = styled.div`
  width: 40%;
  padding: 20px;
  background-color:#14141f;
  ${mobileScreen({ width: "75%" })}
`;

const Logo = styled.h1`
    text-align: center;
    margin-bottom: 30px;
    margin-top: 20px;
    align-items: center;
    justify-content: center;
    cursor: pointer;
    color: #6f50e6;

`


const Box = styled.div`
    border: 1mm solid white;
`

const Title = styled.h1`
  margin: 15px 20px;
  font-size: 24px;
  font-weight: 300;
  color: white;
`;

const Form = styled.form`
  margin: 15px 20px;
  display: flex;
  flex-wrap: wrap;
`;

const Input = styled.input`
  flex: 1;
  min-width: 40%;
  margin: 20px 10px 0px 0px;
  padding: 10px;
`;

const Agreement = styled.span`
  font-size: 12px;
  margin: 20px 0px;
  color: white;
`;

const Button = styled.button`
  width: 40%;
  border: none;
  padding: 15px 20px;
  background-color: #6f50e6;
  color: white;
  cursor: pointer;
`;

function Register() {

  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');
  const [email, setEmail] = useState('');
  const navigate = useNavigate();

  const handleRegister = async (e) => {
    e.preventDefault();
    try{
      const res = await generalRequest.post('/auth/register', {username, email, password});
      navigate('/login');
    } catch(er){
      console.log(er)
    }
  }

  return (
    <Container >
      <Padding>
                <Link style={{textDecoration: 'none'}} to='/'>
                  <Logo>VIRTUAL SPACE</Logo>
                </Link>
        <Box>
            <Title>CREATE ACCOUNT</Title>
            <Form>
            <Input placeholder="Name" />
            <Input placeholder="Last Name" />
            <Input placeholder="Username" onChange={(e)=> setUsername(e.target.value)}/>
            <Input placeholder="Email" onChange={(e)=> setEmail(e.target.value)}/>
            <Input placeholder="Password" onChange={(e)=> setPassword(e.target.value)}/>
            <Input placeholder="Confirm Password" />
            <Agreement>
                By creating an account, I consent to the processing of my personal
                data in accordance with the <b>PRIVACY POLICY</b>
            </Agreement>
            <Button  onClick={handleRegister} >CREATE</Button>
            </Form>
        </Box>
      </Padding>
    </Container>
  )
}

export default Register