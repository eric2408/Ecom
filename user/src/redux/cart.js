import {createSlice} from '@reduxjs/toolkit';


const cart = createSlice({
    name:"cart",
    initialState:{
        products: [],
        quantity:0,
        total:0
    },
    reducers:{
        addProduct: (state, action) => {
            state.quantity += action.payload.quantity;
            state.products.push(action.payload)
            state.total += action.payload.price * action.payload.quantity;
        },
        deleteProduct: (state, action) => {
            state.products= [];
            state.quantity= 0;
            state.total= 0;
        },
        addOne: (state, action) => {
            state.quantity += 1;
            const index = state.products.findIndex(
                (item) => item._id === action.payload.id
            );
            if (index >= 0) {
                state.products[index].quantity += 1;
            }
        },
        subtractOne: (state, action) => {
            const index = state.products.findIndex(
                (item) => item._id === action.payload.id
            );
            if (index >= 0 && state.products[index].quantity > 0) {
                state.products[index].quantity -= 1;
                state.quantity -= 1;
            }
        },
        removeOne:(state, action) => {
            const index = state.products.findIndex(
                    (item) => item._id === action.payload.id
                );
            if (index >= 0) {
                    state.quantity -= state.products[index].quantity;

                    state.products.splice(index, 1);
            } else {
                    console.warn(
                      `Cant remove product (id: ${action.payload.id}) as its not in basket!`
                    )
            }
        }
    }
})

export const {addProduct,  deleteProduct, addOne, subtractOne, removeOne} = cart.actions

export default cart.reducer;

export const emptyCart = async(dispatch) => {
    try{
      dispatch(deleteProduct());
    } catch(e){
      console.log(e)
    }
  }

