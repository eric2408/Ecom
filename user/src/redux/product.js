import { createSlice } from "@reduxjs/toolkit";
import { userRequest } from "../requestAxios";

export const productSlice = createSlice({
  name: "product",
  initialState: {
    products: [],
    isFetching: false,
    error: false,
  },
  reducers: {
    //Get all products
    getProductStart: (state) => {
      state.isFetching = true;
      state.error = false;
    },
    getProductSuccess: (state, action) => {
      state.isFetching = false;
      state.products = action.payload;
    },
    getProductFailure: (state) => {
      state.isFetching = false;
      state.error = true;
    },

    //Delete Products
    deleteProductStart: (state) => {
      state.isFetching = true;
      state.error = false;
    },
    deleteProductSuccess: (state, action) => {
      state.isFetching = false;
      state.products.splice(state.products.findIndex((item) => item._id === action.payload), 1);
    },
    deleteProductFailure: (state) => {
      state.isFetching = false;
      state.error = true;
    },

    //Update products
    updateProductStart: (state) => {
      state.isFetching = true;
      state.error = false;
    },
    updateProductSuccess: (state, action) => {
      state.isFetching = false;
      state.products[
        state.products.findIndex((item) => item._id === action.payload.id)
      ] = action.payload.product;
    },
    updateProductFailure: (state) => {
      state.isFetching = false;
      state.error = true;
    },

    //Add Product
    addProductStart: (state) => {
      state.isFetching = true;
      state.error = false;
    },
    addProductSuccess: (state, action) => {
      state.isFetching = false;
      state.products.push(action.payload);
    },
    addProductFailure: (state) => {
      state.isFetching = false;
      state.error = true;
    },
  },
});

export const {
  getProductStart,
  getProductSuccess,
  getProductFailure,
  deleteProductStart,
  deleteProductSuccess,
  deleteProductFailure,
  updateProductStart,
  updateProductSuccess,
  updateProductFailure,
  addProductStart,
  addProductSuccess,
  addProductFailure,
} = productSlice.actions;

export default productSlice.reducer;

export const getProducts = async (dispatch) => {
    dispatch(getProductStart());
    try {
      const res = await userRequest.get("/products");
      dispatch(getProductSuccess(res.data));
    } catch (err) {
      dispatch(getProductFailure());
    }
  };


export const deleteProduct = async (id, dispatch) => {
  dispatch(deleteProductStart());
  try {
    const res = await userRequest.delete(`/products/${id}`);
    dispatch(deleteProductSuccess(id));
  } catch (err) {
    dispatch(deleteProductFailure());
  }
};

export const updateProduct = async (id, product, dispatch) => {
  dispatch(updateProductStart());
  try {
    const res = await userRequest.put(`/products/${id}`, product);
    dispatch(updateProductSuccess({ id, product }));
  } catch (err) {
    dispatch(updateProductFailure());
  }
};
export const addProduct = async (product, dispatch) => {
  dispatch(addProductStart());
  try {
    const res = await userRequest.post(`/products`, product);
    dispatch(addProductSuccess(res.data));
  } catch (err) {
    dispatch(addProductFailure());
  }
};