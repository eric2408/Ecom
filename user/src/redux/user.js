import { createSlice } from "@reduxjs/toolkit";
import { generalRequest } from "../requestAxios";

const user = createSlice({
  name: "user",
  initialState: {
    currentUser: null,
    isFetching: false,
    error: false,
  },
  reducers: {
    loginStart: (state) => {
      state.isFetching = true;
    },
    loginSuccess: (state, action) => {
      state.isFetching = false;
      state.error = false;
      state.currentUser = action.payload;
    },
    loginFailure: (state) => {
      state.isFetching = false;
      state.error = true;
    },
    logOut: (state) => {
      state.currentUser = null;
    },
  },
});

export const { loginStart, loginSuccess, loginFailure, logOut } = user.actions;
export default user.reducer;


export const login = async(dispatch, user) =>{
    dispatch(loginStart());
    try{
        const res = await generalRequest.post('/auth/login', user)
        dispatch(loginSuccess(res.data))
    } catch(e){
        dispatch(loginFailure())
    }
}

export const logout = async(dispatch, user) => {
  try{
    dispatch(logOut());
  } catch(e){
    console.log(e)
  }
}

