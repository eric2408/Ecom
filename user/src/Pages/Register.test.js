import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import {store, persistor} from '../redux/store';
import{Provider} from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react'
import Register from './Register'

it("matches snapshot", function () {
  const { asFragment } = render(
    <Provider store={store}>
    <PersistGate loading={null} persistor={persistor}>
      <BrowserRouter>
        <Register />
      </BrowserRouter>
    </PersistGate>
  </Provider>

  );
  expect(asFragment()).toMatchSnapshot();
});