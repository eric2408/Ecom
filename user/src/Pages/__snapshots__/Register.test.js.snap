// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe hcSMhq"
  >
    <div
      class="sc-gswNZR iKdlqs"
    >
      <a
        href="/"
        style="text-decoration: none;"
      >
        <h1
          class="sc-dkrFOg kua-DDI"
        >
          VIRTUAL SPACE
        </h1>
      </a>
      <div
        class="sc-hLBbgP iXUAhb"
      >
        <h1
          class="sc-eDvSVe NQZch"
        >
          CREATE ACCOUNT
        </h1>
        <form
          class="sc-jSUZER bWRCyx"
        >
          <input
            class="sc-gKPRtg hnZDxq"
            placeholder="Name"
          />
          <input
            class="sc-gKPRtg hnZDxq"
            placeholder="Last Name"
          />
          <input
            class="sc-gKPRtg hnZDxq"
            placeholder="Username"
          />
          <input
            class="sc-gKPRtg hnZDxq"
            placeholder="Email"
          />
          <input
            class="sc-gKPRtg hnZDxq"
            placeholder="Password"
          />
          <input
            class="sc-gKPRtg hnZDxq"
            placeholder="Confirm Password"
          />
          <span
            class="sc-iBYQkv keDTqX"
          >
            By creating an account, I consent to the processing of my personal data in accordance with the 
            <b>
              PRIVACY POLICY
            </b>
          </span>
          <button
            class="sc-ftTHYK chdYYa"
          >
            CREATE
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
